src/Keystore/EncryptionParameters.h

```diff
@@ -7,34 +7,80 @@
 #pragma once
 
 #include "AESParameters.h"
+#include "Data.h"
 #include "PBKDF2Parameters.h"
 #include "ScryptParameters.h"
-#include "Data.h"
+#include <TrustWalletCore/TWStoredKeyEncryption.h>
 #include <TrustWalletCore/TWStoredKeyEncryptionLevel.h>
 
-#include <variant>
 #include <nlohmann/json.hpp>
 #include <string>
+#include <variant>
 
 namespace TW::Keystore {
 
 /// Set of parameters used when encoding
 struct EncryptionParameters {
-    static EncryptionParameters getPreset(enum TWStoredKeyEncryptionLevel preset) {
+    static TWStoredKeyEncryption getCipher(const std::string& cipher) {
+        if (cipher == "aes-128-ctr") {
```

Comment:
How about a table/map with the 4 (enum, cipher name, key length) tuples, which can be used in the several conversions? Less chance for a typo.

Reply:
Yeah I think there is room for optimization to group all AES parameters directly within the AESParameter Class. The PR aims to solve the aes 256 compatibility only.